annotations/update_pending_annotations.py

```diff
@@ -0,0 +1,152 @@
+        '-t',
+        '--table_id',
+        type=str,
+        default='syn21868591',
+        help=
+        'Synapse table id where annotations will be updated. Probably the Publications Merged table.'
+    )
+    parser.add_argument('-m',
+                        '--manifest_path',
+                        type=str,
+                        help='Path where updated manifest is stored.')
+    return parser.parse_args()
+
+
+def get_updated_df(manifest):
+
+    updated_df = pd.read_csv(manifest, index_col=False).fillna("")
+
+    return updated_df
+
+
+def get_annotations(table_id, updated_df, syn):
+
+    pubmed_list = updated_df['Pubmed Id'].astype(str).tolist()
+    pubmed_string = ', '.join(pubmed_list)
+    print(f'list of pumbmeds to be updated: {pubmed_string}')
+
+    annots_query = syn.tableQuery(
+        f"SELECT pubMedId, assay, tumorType, tissue, dataset, accessibility FROM {table_id} WHERE pubMedId IN ({pubmed_string})"
+    )
+
+    return annots_query
+
+
+def edit_annotations(updated_df, annots_query, syn, table_id):
+
+    # Convert annotations to data frame
+    annots_df = annots_query.asDataFrame().fillna("")
+    # preserve original index
+    index_rows = dict(zip(annots_df.pubMedId, annots_df.index))
+
+    # rename updated_df columns to match annots_df columns
+    updated_df.rename(columns=PUBLICATION_DICT, inplace=True)
+    updated_df.drop(['Component'], axis=1, inplace=True)
+
+    # update annots_df to match updated_df
+    final_df = annots_df.set_index('pubMedId')
+    updated_df = updated_df.set_index('pubMedId')
+    final_df.update(updated_df)
+    # resetting index so pubMedId can be used to map original index
+    final_df.reset_index(inplace=True)
+    # Reset index to match original annots_df using index_rows dict.
+    final_df['index'] = final_df['pubMedId'].map(index_rows)
+    final_df.set_index(keys='index', inplace=True)
+
+    # Get column data types from synapse table
+    cols = syn.getTableColumns(table_id)
+    col_dict = {}
+    for col in cols:
+        for k, v in col.items():
+            if k == 'name':
+                col_dict[v] = col['columnType']
+    data_type_dict = {
+        'STRING': str,
+        'INTEGER': int,
+        'LARGETEXT': str,
+        'STRING_LIST': list,
+        'DOUBLE': float,
+        'LINK': str,
+        'USERID': str,
+        'BOOLEAN': bool
+    }
+    col_types_dict = {k: data_type_dict.get(v, v) for k, v in col_dict.items()}
+
+    # Fix data types in annots_df to match synapse table
+    for columnName in final_df:
+        if col_types_dict[columnName] == list:
+            final_df[columnName] = final_df[columnName].str.split(', ')
+        else:
+            for k, v in col_types_dict.items():
+                final_df[columnName] = final_df[columnName].astype(
+                    col_types_dict[columnName])
+
+    # Can uncomment and examine edited df as a csv before uplaoding.
+    # annots_df.to_csv('updated_annotations.csv', index=False)
```

Comment:
You can alternatively add a `dryrun` flag to your ArgumentParser, then print out the DataFrame whenever it's passed!  e.g.

```
def get_args():
   ...
   parser.add_argument("--dryrun", action="store_true")
   return parser

...

def edit_annotations(updated_df, annots_query, syn, table_id, dryrun):
   ...
   if dryrun:
      annots_df.to_csv('updated_annotations.csv', index=False)
```

That way, you don't have to keep commenting in this line whenever you want to inspect `annots_df`.

Reply:
This is great to know! Thanks you :)